python/lsst/pipe/base/pipeline.py

```diff
@@ -245,11 +245,21 @@ def remove_contracts(label: str):
         # Labels supplied as a set, explicitly remove any that are not in
         # That list
         if labelSpecifier.labels:
+            # update the label specifier to expand any named subsets
+            toRemove = set()
+            toAdd = set()
+            for label in labelSpecifier.labels:
+                if label in pipeline._pipelineIR.named_subsets:
+                    toRemove.add(label)
+                    toAdd.update(pipeline._pipelineIR.named_subsets[label].subset)
```

Comment:
Have you thought about whether this should be recursive (i.e. can subsets nest)?  I don't see a compelling reason why it should be, but it's probably something that should be a conscious choice rather than an implementation artifact.